src/olympia/reviewers/tests/test_utils.py

```diff
@@ -3708,6 +3767,44 @@ def test_reject_multiple_versions_resets_original_status_too(self):
             File.STATUS_DISABLED_REASONS.NONE
         )
 
+    def test_request_legal_review_no_job(self, *, data=None):
+        self.setup_data(
+            amo.STATUS_APPROVED,
+            file_status=amo.STATUS_APPROVED,
+        )
+        if data:
+            data = {**self.get_data(), **data}
+            self.helper.set_data(data)
+        report_request = responses.add(
+            responses.POST,
+            f'{settings.CINDER_SERVER_URL}create_report',
+            json={'job_id': '5678'},
+            status=201,
+        )
+
+        self.helper.handler.request_legal_review()
+
+        assert len(mail.outbox) == 0
+        assert report_request.call_count == 1
+        assert CinderJob.objects.last().job_id == '5678'
+        request_body = json.loads(responses.calls[0].request.body)
+        assert request_body['reasoning'] == 'foo'
+
+    def test_request_legal_review_resolve_job(self):
+        job = CinderJob.objects.create(
+            target_addon=self.addon, resolvable_in_reviewer_tools=True, job_id='1234'
+        )
+        AbuseReport.objects.create(guid=self.addon.guid, cinder_job=job)
+        responses.add_callback(
+            responses.POST,
+            f'{settings.CINDER_SERVER_URL}jobs/1234/decision',
+            callback=lambda r: (201, {}, json.dumps({'uuid': uuid.uuid4().hex})),
+        )
+        self.test_request_legal_review_no_job(data={'cinder_jobs_to_resolve': [job]})
+
+        assert job.reload().decision.action == DECISION_ACTIONS.AMO_LEGAL_FORWARD
+        assert job.forwarded_to_job == CinderJob.objects.last()
```

Comment:
Is there a more explicit, narrow query here that could make use of existing known data?